src/server/actions/workspace-files.actions.ts

```diff
@@ -0,0 +1,40 @@
+'use server'
+
+import * as fs from 'node:fs/promises'
+import * as path from 'node:path'
+import { Action, z } from './action'
+import { DEV_ENV, getConfigValue } from '@/server/config'
+import { getInfoForStudyId } from '@/server/db/queries'
+
+const isMainFile = (filename: string): boolean => {
+    const basename = path.basename(filename, path.extname(filename))
+    return basename.toLowerCase() === 'main'
+}
+
+export const listWorkspaceFilesAction = new Action('listWorkspaceFilesAction', {})
+    .params(z.object({ studyId: z.string() }))
+    .middleware(async ({ params: { studyId } }) => await getInfoForStudyId(studyId))
+    .requireAbilityTo('view', 'Study')
+    .handler(async ({ params: { studyId } }) => {
+        let coderFilesPath = await getConfigValue('CODER_FILES')
+        if (!DEV_ENV) {
+            coderFilesPath += `/${studyId}`
+        }
+        const files = await fs.readdir(coderFilesPath)
```

Comment:
This reads raw directory structures including "directories only", even symlinks theoretically. I think this is not intended. If a pure directory is found here it will break the process I think. Limit it to only files?

Reply:
yes, that is what I meant! Otherwise we might try to read directories later in the process (which will not work)